proposals/devcontainer-templates-distribution.md

```diff
@@ -0,0 +1,120 @@
+
+Each sub-directory should be named such that it matches the `id` field of the `devcontainer-template.json`.  Other files can also be included in the Templates's sub-directory, and will be included during the [packaging step](#packaging) alongside the two required files.  Any files that are not part of the Templates's sub-directory (e.g. outside of `src/dotnet`) will not included in the [packaging step](#packaging).
+
+## Versioning
+
+Each Template is published with only the `latest` tag. Tooling that handles releasing Templates will republish the `latest` tag every time a new release is created.
+
+## Packaging
+
+Templates are distributed as tarballs. The tarball contains the entire contents of the Template sub-directory, including the `devcontainer-template.json`, `.devcontainer.json` (or `.devcontainer/devcontainer.json`), and any other files in the directory.
+
+The tarball is named `devcontainer-template-<id>.tgz`, where `<id>` is the Templates's `id` field.
+
+A reference implementation for packaging and distributing Templates is provided as a GitHub Action (https://github.com/devcontainers/action).
+
+### devcontainer-collection.json
+
+The `devcontainer-collection.json` is an auto-generated metadata file.
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| `sourceInformation` | object | Metadata from the implementing packaging tool. |
+| `templates` | array | The list of Templates that are contained in this collection.|
+
+Each templates's `devcontainer-template.json` metadata file is appended into the `templates` top-level array.
+
+## Distribution
+
+There are several supported ways to distribute templates.  Distribution is handled by the implementing packaging tool.
+
+A user can add the Templates in to their projects as defined by the [Templates Supporting Tools](./devcontainer-templates.md/#adding-a-template-to-a-project-or-codespace).
+
+### OCI Registry
+
+An OCI registry that implements the [OCI Artifact Distribution Specification](https://github.com/opencontainers/distribution-spec) serves as the primary distribution mechanism for Templates.
+
+Each packaged Template is pushed to the registry following the naming convention `<registry>/<namespace>/<id>[:latest]`.
+
+> **Note:** The `namespace` is a unique identifier for the collection of templates and must be different than the collection of [Features](./devcontainer-features.md). There are no strict rules for the `namespace`; however, one pattern is to set `namespace` equal to source repository's `<owner>/<repo>`. 
+
+A custom media type `application/vnd.devcontainers` and `application/vnd.devcontainers.layer.v1+tar` are used as demonstrated below.
```

Comment:
@joshspicer Would it make sense to use a different media type for templates and features to distinguish the two? Does OCI's artifact spec have a recommendation?

Reply:
Thinking of having a `label` to distinguish them, if that makes sense. 

> "labels": {
>     "com.github.package.type": "devcontainer_template"
>   }

For Features we already have it as `"com.github.package.type": "devcontainer_feature"`